src/app/clusters/scenes-server/scenes-server.cpp

```diff
@@ -0,0 +1,874 @@
+
+    // Get Scene Table Instance
+    scenes::SceneTable<scenes::ExtensionFieldSetsImpl> * sceneTable;
+
+    switch (aPath.mAttributeId)
+    {
+    case Attributes::SceneCount::Id:
+        sceneTable = scenes::GetSceneTableImpl(aPath.mEndpointId, endpointTableSize);
+        ReturnErrorOnFailure(sceneTable->GetEndpointSceneCount(value));
+        return aEncoder.Encode(value);
+    case Attributes::RemainingCapacity::Id:
+        sceneTable = scenes::GetSceneTableImpl(aPath.mEndpointId, endpointTableSize);
+        ReturnErrorOnFailure(sceneTable->GetRemainingCapacity(aEncoder.AccessingFabricIndex(), value));
+        return aEncoder.Encode(value);
+    default:
+        return CHIP_NO_ERROR;
+    }
+}
+
+void ScenesServer::GroupWillBeRemoved(FabricIndex aFabricIx, EndpointId aEndpointId, GroupId aGroupId)
+{
+    // Get Scene Table Instance
+    scenes::SceneTable<scenes::ExtensionFieldSetsImpl> * sceneTable = scenes::GetSceneTableImpl(aEndpointId);
+    VerifyOrReturn(nullptr != sceneTable);
+
+    MakeSceneInvalid(aEndpointId);
+
+    VerifyOrReturn(nullptr != mGroupProvider);
+    if (0 != aGroupId && !mGroupProvider->HasEndpoint(aFabricIx, aGroupId, aEndpointId))
+    {
+        return;
+    }
+
+    sceneTable->DeleteAllScenesInGroup(aFabricIx, aGroupId);
+}
+
+void ScenesServer::MakeSceneInvalid(EndpointId aEndpointId)
+{
+    Attributes::SceneValid::Set(aEndpointId, false);
+}
+
+void ScenesServer::StoreCurrentScene(FabricIndex aFabricIx, EndpointId aEndpointId, GroupId aGroupId, SceneId aSceneId)
+{
+    if (CHIP_NO_ERROR == StoreSceneParse(aFabricIx, aEndpointId, aGroupId, aSceneId, mGroupProvider))
+    {
+        Attributes::SceneValid::Set(aEndpointId, true);
+    }
+}
+void ScenesServer::RecallScene(FabricIndex aFabricIx, EndpointId aEndpointId, GroupId aGroupId, SceneId aSceneId)
+{
+    VerifyOrReturn(EMBER_ZCL_STATUS_SUCCESS == Attributes::SceneValid::Set(aEndpointId, false));
+
+    Optional<DataModel::Nullable<uint16_t>> transitionTime;
+
+    if (CHIP_NO_ERROR == RecallSceneParse(aFabricIx, aEndpointId, aGroupId, aSceneId, transitionTime, mGroupProvider))
+    {
+        Attributes::SceneValid::Set(aEndpointId, true);
+    }
+}
+
+void ScenesServer::HandleAddScene(HandlerContext & ctx, const Commands::AddScene::DecodableType & req)
+{
+    AddSceneParse<Commands::AddScene::DecodableType, Commands::AddSceneResponse::Type>(ctx, req, mGroupProvider);
+}
+
+void ScenesServer::HandleViewScene(HandlerContext & ctx, const Commands::ViewScene::DecodableType & req)
+{
+    ViewSceneParse<Commands::ViewScene::DecodableType, Commands::ViewSceneResponse::Type>(ctx, req, mGroupProvider);
+}
+
+void ScenesServer::HandleRemoveScene(HandlerContext & ctx, const Commands::RemoveScene::DecodableType & req)
+{
+    Commands::RemoveSceneResponse::Type response;
+
+    uint16_t endpointTableSize = 0;
+    ReturnOnFailure(AddResponseOnError<Commands::RemoveSceneResponse::Type>(
```

Comment:
You still have a bunch of these unnecessary explicit types listed, here and in various functions below.